From 9f1012b580c89b1f42c1beeda122680f831e048e Mon Sep 17 00:00:00 2001
From: manvithapula <manvithapula@gmail.com>
Date: Thu, 27 Jun 2024 22:02:01 +0530
Subject: [PATCH 10/11] commit

---
 client3.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/client3.py b/client3.py
index 3fc09b7..c5fa683 100644
--- a/client3.py
+++ b/client3.py
@@ -36,7 +36,6 @@ def getDataPoint(quote):
     stock = quote['stock']
     bid_price = float(quote['top_bid']['price'])
     ask_price = float(quote['top_ask']['price'])
+    price = (bid_price+ask_price)/2
-    price = bid_price
     return stock, bid_price, ask_price, price
 
@@ -42,7 +43,9 @@ def getDataPoint(quote):
 def getRatio(price_a, price_b):
     """ Get ratio of price_a and price_b """
     """ ------------- Update this function ------------- """
-    return 1
+    if price_b == 0:
+        return 
+    return price_a / price_b
 
 
 # Main
@@ -52,8 +55,10 @@ if __name__ == "__main__":
         quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
         """ ----------- Update to get the ratio --------------- """
+        prices = {}
         for quote in quotes:
             stock, bid_price, ask_price, price = getDataPoint(quote)
+            prices[stock] = price
             print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-        print("Ratio %s" % getRatio(price, price))
+        print("Ratio %s" % getRatio(price["ABC"], price["DEF"]))
-- 
2.39.1.windows.1
